transports/realtime-websocket-transport/src/pipecatWebSocketTransport.ts

```diff
@@ -0,0 +1,225 @@
+import { RealTimeWebsocketTransport } from "./realTimeWebSocketTransport.ts";
+import { DailyMediaManager } from "../../../lib/media-mgmt/dailyMediaManager.ts";
+
+import { Frame } from "./generated/proto/frames.ts";
+
+export class PipecatWebsocketTransport extends RealTimeWebsocketTransport {
+  declare private _ws: ReconnectingWebSocket | null;
+  declare private _botResponseID: number;
+  private static RECORDER_SAMPLE_RATE = 16_000;
+  private audioQueue: ArrayBuffer[] = [];
+
+  constructor() {
+    let mediaManager = new DailyMediaManager(
+      true,
+      true,
+      undefined,
+      undefined,
+      512,
+      PipecatWebsocketTransport.RECORDER_SAMPLE_RATE,
+    );
+    super({}, mediaManager);
+    this._ws = null;
+    this._botResponseID = 0;
+  }
+
+  async initializeLLM(): Promise<void> {}
+
+  attachLLMListeners(): void {
+    // nothing to do here
+  }
+
+  initializeWebsocket(authBundle: any): void {
+    this._ws = new ReconnectingWebSocket(`${authBundle.ws_url}`, undefined, {
+      parseBlobToJson: false,
+    });
+    // disabling the keep alive, there is no API for it inside Pipecat
+    this._ws.keepAliveInterval = 0;
+    if (!this._ws) {
+      logger.error(
+        "attachLLMListeners called before the websocket is initialized. Be sure to call initializeLLM() first.",
+      );
+      return;
+    }
+    this._ws.on("open", () => {
+      logger.debug("Websocket connection opened");
+    });
+    this._ws.on("message", async (data: Blob) => {
+      let arrayBuffer: ArrayBuffer = await data.arrayBuffer();
+      const parsedFrame = Frame.fromBinary(new Uint8Array(arrayBuffer)).frame;
+      if (parsedFrame.oneofKind === "audio") {
+        if (!this._botIsSpeaking) {
+          this._botResponseID++;
+          this.botStartedSpeaking();
+        }
+        // We should be able to use parsedFrame.audio.audio.buffer but for
+        // some reason that contains all the bytes from the protobuf message.
+        const audioVector = Array.from(parsedFrame.audio.audio);
+        const uint8Array = new Uint8Array(audioVector);
+        const int16Array = new Int16Array(uint8Array.buffer);
+        this.bufferBotAudio(int16Array, this._botResponseID.toString());
+      } else if (parsedFrame.oneofKind === "message") {
+        let jsonText = parsedFrame.message.data;
+        try {
+          let jsonMessage = JSON.parse(jsonText);
+          if (jsonMessage.label === "rtvi-ai") {
+            this._onMessage(jsonMessage as RTVIMessage);
+          }
+        } catch {
+          logger.warn("Failed to parse message", jsonText);
+        }
+      }
+    });
+    this._ws.on("error", (error: Error) => {
+      this.connectionError(`websocket error: ${error}`);
+    });
+    this._ws.on("connection-timeout", () => {
+      this.connectionError("websocket connection timed out");
+    });
+    this._ws.on("close", (code: number) => {
+      this.connectionError(`websocket connection closed. Code: ${code}`);
```

Comment:
this is triggering an error, but is close always an error? what about on actual disconnect?